web-api/terraform/template/elasticsearch.tf

```diff
@@ -49,3 +49,27 @@ resource "aws_elasticsearch_domain" "efcms-search" {
     log_type                 = "ES_APPLICATION_LOGS"
   }
 }
+
+resource "aws_elasticsearch_domain" "efcms-app-logs" {
+  domain_name           = "efcms-app-logs-${var.environment}"
+  elasticsearch_version = "7.4"
+
+  cluster_config {
+    instance_type = "t2.small.elasticsearch"
+    instance_count = var.es_instance_count == "" ? "1" : var.es_instance_count
+  }
+
+  ebs_options{
+    ebs_enabled = true
+    volume_size = 10
+  }
+
+  snapshot_options {
+    automated_snapshot_start_hour = 23
+  }
+
+  log_publishing_options {
+    cloudwatch_log_group_arn = aws_cloudwatch_log_group.elasticsearch_application_logs.arn
```

Comment:
We’ll probably want a separate log group for these logs — I’m worried we might get into an infinite loop if we intermingle the application’s elasticsearch logs with the logging logs (new log stream item -> elasticsearch -> log stream -> new log stream item -> elasticsearch…).

… does that make sense? I used the word "log" so many times :joy:

Reply:
I wondered about this too - it seems potentially confusing if nothing else! I'll separate them out.